packages/devextreme/js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -1434,8 +1468,12 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
         const { clickEventNamespace, itemSelector, pointerDownEventNamespace, nodeSelector } = this._getItemClickEventData();
 
         eventsEngine.on($itemContainer, clickEventNamespace, itemSelector, (e) => {
+            if($(e.target).hasClass(CHECK_BOX_ICON_CLASS) || $(e.target).hasClass(CHECK_BOX_CLASS)) {
```

Comment:
Can we use .closest(CHECK_BOX_CLASS) here?

Reply:
1 of them is CHECK_BOX_ICON_CLASS, not CHECK_BOX_CLASS

Reply:
I understand that this is necessary, since you e.target will be one of 2 elements - root or some kind of internal. For this case, the check should be suitable - $(e.target).closest(CHECK_BOX_CLASS), i.e. here we get rid of the knowledge of the insides of what is inside the root container and what elements there may be from those that were clicked and captured the click